cdc/sink/dmlsink/txn/mysql/mysql.go

```diff
@@ -652,15 +653,29 @@ func (s *mysqlBackend) multiStmtExecute(
 	ctx context.Context, dmls *preparedDMLs, tx *sql.Tx, writeTimeout time.Duration,
 ) error {
 	start := time.Now()
-	multiStmtSQL := ""
-	multiStmtArgs := []any{}
+	var (
+		sb            strings.Builder
+		multiStmtArgs []any
+	)
+	// preallocate the string builder's capacity to avoid memory reallocation.
+	size := 0
 	for i, query := range dmls.sqls {
-		multiStmtSQL += query
+		size += len(query)
 		if i != len(dmls.sqls)-1 {
-			multiStmtSQL += ";"
+			size += 1
+		}
+	}
+	sb.Grow(size)
+
+	for i, query := range dmls.sqls {
```

Comment:
Why not use `strings.Join`?

Reply:
it looks that `strings.Join` also use the string builder, it can be replaced by using it, let me fix it.

Reply:
It looks that the `strings.Join` does not consider the `separator` when calculate the string builder capacity before the `Grow` method, it may cause reallocation.

Reply:
Yes, `strings.Join` can be used in this case.